sentinelhub/download/client.py

```diff
@@ -89,16 +88,18 @@ def download(
         :param download_requests: A list of requests or a single request to be executed.
```

Comment:
Are `DownloadRequest` and `DownloadResponse` ever specialized in when subclassing download client? If so, I believe it would be nice if we turn it into a generic. This can (probably should) be done in a separate MR

Reply:
Specialized in terms of what type of data they will download and decode? There are actually a lot of options, usually it depends on `data_type` parameter although, if not provided, the type can also be decided from response headers.

So it would be really tricky to turn this into generics.

Reply:
specialized more as 'subclassed'. So if we ever have `BolhaDownloadClient` with `BolhaDownloadResponse` instead of regular just the `DownloadResponse`.

Reply:
Sure.